src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs

```diff
@@ -202,66 +206,53 @@ private void IgnoreUnsupportProjectReference(PackageSpec project)
         /// <summary>
         /// Gets the installed (top level) package references for this project. 
         /// </summary>
-        public async override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
+        public override async Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
         {
             ProjectPackages packages = await GetInstalledAndTransitivePackagesAsync(token);
             return packages.InstalledPackages;
         }
 
-        /// <summary>
-        /// Gets the both the installed (top level) and transitive package references for this project.
-        /// Returns the package reference as two separate lists (installed and transitive).
-        /// </summary>
-        public override async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(CancellationToken token)
+        /// <inheritdoc/>
+        public override async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(IList<LockFileTarget> existingTargets, CancellationToken token)
```

Comment:
Can `GetInstalledAndTransitivePackagesAsync` from legacy and Cps PR be merged? 

Would simplify the code significantly.

They seem to large equivalently other than the fact that the CPS version has a null check. 

Adding a similar check wouldn't make the legacy code wrong I think,.

Reply:
Ok! I will try to merge both implementations

Reply:
Will resolve in follow-up issue: https://github.com/NuGet/Home/issues/11397